polyfills/_ESAbstract/Construct/polyfill.js

```diff
@@ -0,0 +1,29 @@
+/* global IsConstructor, OrdinaryCreateFromConstructor, Call */
+// 7.3.13. Construct ( F [ , argumentsList [ , newTarget ]] )
+function Construct(F /* [ , argumentsList [ , newTarget ]] */) { // eslint-disable-line no-unused-vars
+	// 1. If newTarget is not present, set newTarget to F.
+	var newTarget = arguments.length > 2 ? arguments[2] : F;
+
+	// 2. If argumentsList is not present, set argumentsList to a new empty List.
+	var argumentsList = arguments.length > 1 ? arguments[1] : [];
+
+	// 3. Assert: IsConstructor(F) is true.
+	if (!IsConstructor(F)) {
+		throw new TypeError('F must be a constructor.');
+	}
+
+	// 4. Assert: IsConstructor(newTarget) is true.
+	if (!IsConstructor(newTarget)) {
+		throw new TypeError('newTarget must be a constructor.');
+	}
+
+	// 5. Return ? F.[[Construct]](argumentsList, newTarget).
+	// Polyfill.io - If newTarget is the same as F, it is equivalent to new F(...argumentsList).
+	if (newTarget === F) {
+		return new (Function.prototype.bind.apply(F, [null].concat(argumentsList)))();
```

Comment:
Mind blowing. 😁 
Note to self: `apply` is acting on `bind`. `null` in the arguments list becomes the first argument `thisArg` to bind. https://tc39.github.io/ecma262/#sec-function.prototype.bind